metarecord/models/attribute.py

```diff
@@ -64,15 +64,24 @@ def is_free_text(self):
 class AttributeValue(TimeStampedModel, UUIDPrimaryKeyModel):
     attribute = models.ForeignKey(Attribute, verbose_name=_('attribute'), related_name='values')
     value = models.CharField(verbose_name=_('value'), max_length=1024)
+    index = models.PositiveSmallIntegerField(db_index=True)
 
     class Meta:
         verbose_name = _('attribute value')
         verbose_name_plural = _('attribute values')
-        unique_together = ('attribute', 'value')
+        unique_together = (('attribute', 'value'), ('attribute', 'index'))
+        ordering = ('index',)
 
     def __str__(self):
         return self.value
 
+    def save(self, *args, **kwargs):
+        if not self.index:
+            self.index = max(
+                AttributeValue.objects.filter(attribute=self.attribute).values_list('index', flat=True) or [0]
```

Comment:
That is actually copied from Attribute, which has it copied originally from KerroKantasi IIRC :)

What about:

```
last_index = AttributeValue.objects.filter(attribute=self.attribute).values_list('index').last()
self.index = last_index + 1 if last_index else 1
```

And yes, it can cause a race condition in theory, but in practise that shouldn't be a problem because those indexes should be updated very rarely and not via the API. And it doesn't cause any real harm anyway. Could have added a comment about that though.